Jenkinsfile

```diff
@@ -25,33 +25,60 @@ pipeline {
     // Right now, we're *only* building and deploying on the `master` branch;
     // longer-term, we'll want to deploy feature branches as well.
 
-    stage('Build base Docker image') {
+    stage('Build staging Docker image') {
+      when {
+        branch 'master'
```

Comment:
Remove this check if we want to continue running test builds on every PR, then wrap the push to DockerHub in a conditional.